YellowNotes.Core/Repositories/CategoryRepository.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using YellowNotes.Core.Dtos;
+using YellowNotes.Core.Models;
+
+namespace YellowNotes.Core.Repositories
+{
+    public class CategoryRepository : ICategoryRepository
+    {
+        private readonly DatabaseContext context;
+
+        private readonly IMapper mapper;
+
+        private readonly int maxCategoryCount = 10;
```

Comment:
why is that? :O

Reply:
Mainly because of query string length and front-end design.